boa/verifiers.py

```diff
@@ -166,6 +185,24 @@ def verify(
         address=contract.address,
         solc_json=bundle,
         contract_name=contract.contract_name,
-        license_type=license_type,
+        constructor_calldata=contract.ctor_calldata,
         wait=wait,
+        **kwargs,
     )
+
+
+def _wait_until(
+    predicate: Callable[[], T],
+    wait_for: timedelta,
+    backoff: timedelta,
+    backoff_factor: float,
+) -> T:
+    timeout = datetime.now() + wait_for
+    wait_time = backoff
+    while datetime.now() < timeout:
+        if result := predicate():
+            return result
```

Comment:
why return the result?

Reply:
so it could be used with more than just bools (but yeah I know you don't like truthiness)

Reply:
So the value is actually used to wait until the contract is created:
https://github.com/vyperlang/titanoboa/blob/a83112eb87c7d20525301826409ca1eb85cce04b/boa/explorer.py#L104